unlock-app/src/components/interface/checkout/alpha/index.tsx

```diff
@@ -25,7 +25,7 @@ export function CheckoutPage() {
   const paywallConfig = communication.paywallConfig || paywallConfigFromQuery
 
   const injectedProvider =
-    communication.providerAdapter || selectProvider(config)
+    selectProvider(config) || communication.providerAdapter
```

Comment:
Actually the order here is important! Let's reverse that.
This is because we should let the main frame "force" a wallet :)